elasticsearch/elasticsearch.go

```diff
@@ -258,77 +306,70 @@ func (db *Database) PostDocument(index, type_ string, doc interface{}) (string,
+// PutDocumentVersion creates or updates the document in the given index
+// if the version parameter is the same as the currently stored version.
+// The type_ parameter controls how the document will be indexed.
+// PutDocumentVersion returns an error with a cause that satisfies
+// IsConflictError if the data cannot be stored due to a version
+// mismatch, and a non-nil error if any other error occurs. See
+// http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/docs-index_.html#index-versioning
 // for more information.
 func (db *Database) PutDocumentVersion(index, type_, id string, version int64, doc interface{}) error {
-	return db.PutDocumentVersionWithType(index, type_, id, version, "internal", doc)
+	return errgo.Mask(db.PutDocumentVersionWithType(index, type_, id, version, "internal", doc), IsElasticsearchError)
 }
 
-// PutDocumentVersion creates or updates the document in the given index if the version
-// parameter is the same as the currently stored version. The type_ parameter
-// controls how the document will be indexed. PutDocumentVersionWithType returns
-// ErrConflict if the data cannot be stored due to a version mismatch, and a non-nil error if
-// any other error occurs.
+// PutDocumentVersionWithType creates or updates the document in the
+// given index if the version parameter is the same as the currently
+// stored version. The type_ parameter controls how the document will be
+// indexed. PutDocumentVersionWithType returns an error with a cause that
+// satisfies IsConflictError if the data cannot be stored due to a
+// version mismatch, and a non-nil error if any other error occurs.
 //
-// The constants Internal, External and ExternalGTE represent some of the available
-// version types. Other version types may also be available, plese check the elasticsearch
-// documentation.
+// The constants Internal, External and ExternalGTE represent some of the
+// available version types. Other version types may also be available,
+// plese check the elasticsearch documentation.
 //
-// See http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/docs-index_.html#index-versioning
-// and http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/docs-index_.html#_version_types for more information.
+// See
+// http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/docs-index_.html#index-versioning
+// and
+// http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/docs-index_.html#_version_types
+// for more information.
 func (db *Database) PutDocumentVersionWithType(
 	index, type_, id string,
 	version int64,
 	versionType string,
 	doc interface{}) error {
 	url := fmt.Sprintf("%s?version=%d&version_type=%s", db.url(index, type_, id), version, versionType)
-	if err := db.put(url, doc, nil); err != nil {
-		return getError(err)
-	}
-	return nil
+	return errgo.Mask(db.put(url, doc, nil), IsElasticsearchError)
```

Comment:
You could potentially avoid all these explicit Mask calls by doing:


    var mask = errgo.MaskFunc(IsElasticSearchError)

then:

    return mask(db.put(url, doc, nil))

etc
*shrug*